tests/ergo_run_test.go

```diff
@@ -542,3 +551,223 @@ func TestRunOSX(t *testing.T) {
+	startDynamicTestWebServer()
+
+	//run ergo
+	cmd := ergo("run")
+	defer func() {
+		if cmd == nil {
+			os.Exit(1)
+			return
+		}
+
+		if cmd.Process != nil {
+			cmd.Process.Kill()
+		}
+	}()
+
+	go func() {
+		cmd.Run()
+	}()
+
+	//wait for it to be sure it started
+	time.Sleep(2 * time.Second)
+
+	//we should not have a correct response
+	rez, err := getCommandResult("powershell", "-c", "$proxy=[System.Net.WebRequest]::GetSystemWebProxy();",
+		"$webclient=new-object system.net.webclient;", "$webclient.proxy=$proxy;",
+		"$webclient.DownloadString('http://dynamic.dev')")
+
+	//on windows we should get an error here
+	if err == nil {
+		t.Fatal("Expected error while requesting http://dynamic.dev, Got nil\r\n")
+	}
+
+	//make sure that we rewrite the config with the original values
+	fileContent, err := ioutil.ReadFile("./.ergo")
+
+	if err == nil {
+		//we clean after the test. Otherwise the next test will fail
+		defer ioutil.WriteFile("./.ergo", fileContent, 0755)
+	}
+
+	f, err := os.OpenFile("./.ergo", os.O_APPEND|os.O_WRONLY, 0600)
+	if err != nil {
+		t.Fatal("Could not write on the config file")
+	}
+
+	defer f.Close()
+
+	if _, err = f.WriteString("dynamic http://localhost:9090"); err != nil {
+		panic(err)
+	}
+
+	time.Sleep(1 * time.Second)
+
+	rez, err = getCommandResult("powershell", "-c", "$proxy=[System.Net.WebRequest]::GetSystemWebProxy();",
+		"$webclient=new-object system.net.webclient;", "$webclient.proxy=$proxy;",
+		"$webclient.DownloadString('http://dynamic.dev')")
+
+	if err != nil {
+		t.Fatalf("Expected no error while asking for an added config. Got %s\r\n", err.Error())
+	}
+
+	if strings.Trim(rez, " \r\n") != "ergo test response" {
+		t.Fatalf("Expected \"ergo test response\" as response and got: %s\r\n", rez)
+	}
+}
+
+func TestConfigDynamicLinuxGnome(t *testing.T) {
+	if *getOS() != "linux-gnome" {
+		t.Skip("Not running linux-gnome run specific tests")
+	}
+	err := setupErgo()
+	//exit status 1 means that gsettings was not found. So if linux is not really gnome, this will fails otherwise
+	if err != nil && err.Error() != "exit status 1" {
+		t.Fatalf("Could not perform setup for linux-gnome. Got %s", err.Error())
+	}
+
+	defer cleanSetup()
+
+	startDynamicTestWebServer()
+
+	cmd := ergo("run")
+	if cmd != nil {
+		defer func() {
+			if cmd != nil && cmd.ProcessState != nil && !cmd.ProcessState.Exited() {
+				cmd.Process.Kill()
+			}
+		}()
+	}
+	go func() {
+		cmd.Run()
+	}()
+
+	time.Sleep(2 * time.Second)
+
+	//it would be better perhaps to use a brower like chrome-headless
+	rez, err := getCommandResult("./testCurlNotExisting.sh")
+
+	if err != nil {
+		t.Fatalf("Expected no error while requesting http://dynamic.dev, Got %s\r\n", err.Error())
+	}
+
+	if rez == "ergo test response" {
+		t.Fatalf("Expected diffrent response while asking for http://dynamic.dev, Got %s\r\n", rez)
+	}
+
+	//make sure that we rewrite the config with the original values
+	fileContent, err := ioutil.ReadFile("./.ergo")
```

Comment:
I noticed this snippet occurs quite often. How about creating a helper function? I think it may make the test clean and focus only in what it is testing that is `when running ergo we should receive a response x`.

Reply:
Yep. This is also true. Perhaps in another PR (with the previous one)?

Reply:
Sure agreed! We can merge and ask for a refactor later